app/views/vendors/_vendor_header.html.erb

```diff
@@ -0,0 +1,47 @@
+<header class="navbar navbar-fixed-top navbar-inverse">
+  <div class="container">
+    <%= link_to "Hi, #{session[:vendor_name]}", vendor_path(session[:vendor_id]), :id => "logo" %>
+    <nav>
+      <ul class="nav navbar-nav navbar-right">
+        <li><%= link_to "Home", vendor_path(session[:vendor_id]) %></li>
+        <li class="dropdown">
+          <a href="#" class="dropdown-toggle" data-toggle="dropdown">
+            Products & Sales <b class="caret"></b>
+          </a>
+          <ul class="dropdown-menu">
+            <li><%= link_to "Create Product", new_vendor_product_path(session[:vendor_id]) %></li>
+            <li class="divider"></li>
+            <li><%= link_to "Add Sale", vendor_products_path(session[:vendor_id]) %></li>
+          </ul>
+        </li>
+        <li class="dropdown">
+          <a href="#" class="dropdown-toggle" data-toggle="dropdown">
+            Sales Stats <b class="caret"></b>
+          </a>
+          <ul class="dropdown-menu">
+            <li><%= link_to "Total Sales", sales_vendor_path(session[:vendor_id]) %></li>
+            <li class="divider"></li>
+            <li class="dropdown-header">
+              <a href="#" class="dropdown-toggle" data-toggle="dropdown">
+                Monthly Sales
+              </a>
+            </li>
+            <li><%= link_to "January", sales_month_vendor_path(session[:vendor_id], 1) %></li>
```

Comment:
I wonder if you could put this `link_to` code in a partial and use locals in order to dynamically generate these